src/sonic-yang-models/yang-models/sonic-pbh.yang

```diff
@@ -0,0 +1,218 @@
+					type leafref {
+						path "/pbh:sonic-pbh/pbh:PBH_TABLE/pbh:PBH_TABLE_LIST/pbh:table_name";
+					}
+				}
+
+				leaf rule_name {
+					description "The name of this rule";
+					type string;
+				}
+
+				leaf priority {
+					description "Configures priority for this rule";
+					mandatory true;
+					type uint32;
+				}
+
+				leaf gre_key {
+					description "Configures packet match for this rule: GRE key";
+					type string {
+						pattern "(0x){1}[a-fA-F0-9]+/(0x){1}[a-fA-F0-9]+";
+					}
+				}
+
+				leaf ip_protocol {
+					description "Configures packet match for this rule: IP protocol";
+					type string {
+						pattern "(0x){1}[a-fA-F0-9]+/(0x){1}[a-fA-F0-9]+";
+					}
+				}
+
+				leaf ipv6_next_header {
+					description "Configures packet match for this rule: IPv6 Next header";
+					type string {
+						pattern "(0x){1}[a-fA-F0-9]+/(0x){1}[a-fA-F0-9]+";
+					}
+				}
+
+				leaf l4_dst_port {
+					description "Configures packet match for this rule: L4 destination port";
+					type string {
+						pattern "(0x){1}[a-fA-F0-9]+/(0x){1}[a-fA-F0-9]+";
+					}
+				}
+
+				leaf inner_ether_type {
+					description "Configures packet match for this rule: inner EtherType";
+					type string {
+						pattern "(0x){1}[a-fA-F0-9]+/(0x){1}[a-fA-F0-9]+";
+					}
+				}
+
+				leaf-list hash_list {
+					description "The list of hash fields to apply with this rule";
+					min-elements 1;
+					type leafref {
+						path "/pbh:sonic-pbh/pbh:PBH_HASH/pbh:PBH_HASH_LIST/pbh:hash_name";
+					}
+				}
+
+				leaf packet_action {
+					description "Configures packet action for this rule";
+					type string {
+						pattern "SET_ECMP_HASH|SET_LAG_HASH";
+					}
+					default "SET_ECMP_HASH";
+				}
+
+				leaf flow_counter {
+					description "Enables/Disables packet/byte counter for this rule";
+					type string {
+						pattern "DISABLED|ENABLED";
+					}
+					default "DISABLED";
+				}
+
+			}
+			/* end of PBH_RULE_LIST */
+		}
+		/* end of container PBH_RULE */
+
+		container PBH_TABLE {
+
+			description "PBH_TABLE part of config_db.json";
+
+			list PBH_TABLE_LIST {
+
+				key "table_name";
+
+				leaf table_name {
+					description "The name of this table";
+					type string;
+				}
+
+				leaf-list port_list {
+					description "Ports to which this table is applied";
+					type union {
+						type leafref {
+							path "/port:sonic-port/port:PORT/port:PORT_LIST/port:name";
+						}
+						type string {
+							pattern "";
+						}
+					}
+					default "";
+				}
+
+				leaf-list lag_list {
+					description "Portchannels to which this table is applied";
+					type union {
+						type leafref {
+							path "/lag:sonic-portchannel/lag:PORTCHANNEL/lag:PORTCHANNEL_LIST/lag:name";
+						}
+						type string {
+							pattern "";
+						}
+					}
+					default "";
+				}
+
+				leaf description {
+					description "The description of this table";
+					mandatory true;
+					type string {
+						length 0..255;
```

Comment:
What does this field contain when len is 0?

Reply:
@venkatmahalingam empty string. It's by design.

Reply:
not sure how the empty string helps in the description, can we make this field optional?

Reply:
@venkatmahalingam ok. We can make it optional.